### DISCUSSION TITLE
Privileged port forwarding does not work on Linux

### DISCUSSION THREAD

**Discussion by dmarkey** [lima-vm/lima#1524] [General]
Hey, I'm trying to create a k3s solution that works on mac (amd64/arm64 and Linux). Lima VM seems like a good fit. However I need to be able have port 80 and 443 forwarding. This works find on Mac with 0.0.0.0 as the bind address but not Linux.

Any ideas or workarounds?

**Comment by jandubois** [comment]
Run `sudo sysctl -w net.ipv4.ip_unprivileged_port_start 80` to allow Lima to bind to those ports.

**Comment by dmarkey** [comment]
Thanks so much, is there any configuration option to allow scripts to run on the host to set this up like https://github.com/phinze/vagrant-host-shell?

I also need to set additional loopback addresses.

**Comment by jandubois** [comment]
No, there is no built-in mechanism that allows the VM to execute commands on the host. You would have to run an ssh daemon on the host and provide the VM with the keys to connect back to it.

Note that the `sysctl` cohas to be run as root, so will either prompt you for the `sudo` password, or you need to setup password-less `sudo`. So it feels quite risky to allow the VM to do this automatically; I think I would rather wrap the `limactl start` command in a shell script that does the host setup there.
